packages/graphql/src/schema/create-relationship-fields/create-relationship-fields.ts

```diff
@@ -121,6 +120,24 @@ function createRelationshipFields({
             const nodeFieldInputName = `${rel.connectionPrefix}${upperFieldName}FieldInput`;
             nodeFieldInput = schemaComposer.getOrCreateITC(nodeFieldInputName);
         }
+        // Don't generate an empty input type
+        let nodeFieldUpdateInput: InputTypeComposer<any> | undefined;
+        if (
+            nestedOperations.size !== 0 &&
+            // If the only nestedOp is connectOrCreate, it won't be generated if there are no unique fields on the related type
+            !(
+                nestedOperations.size === 1 &&
+                nestedOperations.has(RelationshipNestedOperationsOption.CONNECT_OR_CREATE) &&
+                !node.uniqueFields.length
+            )
+        ) {
```

Comment:
What do you think about writing this as a declared meaningful variable? I'm thinking like `onlyConnectOrCreateAndNoUniqueFields`, so it would be like this:

```suggestion
        let nodeFieldUpdateInput: InputTypeComposer<any> | undefined;
        // If the only nestedOperation is connectOrCreate, it won't be generated if there are no unique fields on the related type
        const onlyConnectOrCreateAndNoUniqueFields =
            nestedOperations.size === 1 &&
            nestedOperations.has(RelationshipNestedOperationsOption.CONNECT_OR_CREATE) &&
            !node.uniqueFields.length;

        if (
            nestedOperations.size !== 0 &&
            !(onlyConnectOrCreateAndNoUniqueFields)
        ) {
```